### ISSUE TITLE
Version 24.6.18.1 powershell not found during WinPE boot

### ISSUE THREAD

**Issue by rroethof** [OSDeploy/OSD#236]
As I have seen in OSDeploy/OSD#239 I am experiencing similar issues.
The WinPE boot is missing powershell completely.

In the command prompt I tried:
cd c:\windows\system32\WindowsPowerShell\v1.0
Then did a 'dir' command and it only shows the directory 'Modules'
Powershell.exe seems to be missing completely

**Comment by jkingster** [comment]
I'm experiencing the same issue. Utilizing a fresh 24H2 install, along with the most recent WinRE/ADK pulled from here: https://learn.microsoft.com/en-us/windows-hardware/get-started/adk-install

Same versions.

**Comment by OSDeploy** [comment]
> I'm experiencing the same issue. Utilizing a fresh 24H2 install, along with the most recent WinRE/ADK pulled from here: https://learn.microsoft.com/en-us/windows-hardware/get-started/adk-install
> 
> Same versions.

Same answer as above:

"Review your logs for errors during the Template build to verify in C:\ProgramData\OSDCloud\Logs\Template"

**Comment by jkingster** [comment]
Sorry, forgot to post them.  I see nothing out of the ordinary besides a good chunk of the following:

`2025-02-28-165458 Save Windows Image
Dism Function: Save-WindowsImage
PS>TerminatingError(Save-WindowsImage): "Cannot validate argument on parameter 'Path'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again."

PS>TerminatingError(Add-WindowsPackage): "Cannot bind argument to parameter 'Path' because it is null."
C:\ProgramData\OSDCloud\Logs\Template\2025-02-28-165456-Add-WindowsPackage-WinPE-StorageWMI_en-us.log
C:\Program Files (x86)\Windows Kits\10\Assessment and Deployment Kit\Windows Preinstallation Environment\AMD64\WinPE_OCs\en-us\WinPE-WDS-Tools_en-us.cab
PS>TerminatingError(Add-WindowsPackage): "Cannot bind argument to parameter 'Path' because it is null."
C:\ProgramData\OSDCloud\Logs\Template\2025-02-28-165456-Add-WindowsPackage-WinPE-WDS-Tools_en-us.log`

I can post the entire .log file if necessary

**Comment by OSDeploy** [comment]
Please make a first pass and see if you can narrow down what the issue is. Unless others are having this problem, it's probably not something I can replicate.

Also, make sure you install the ADK WinPE Add-on.

David

**Comment by jkingster** [comment]
Yes, the ADK WinPE add-on is installed and matches the ADK version. Looking at the start of the .log, the initial terminating error happens when `Mount-WindowImage` seems to be called. 

```
2025-02-28-171806 Mount Boot.wim
OSD Function: Mount-MyWindowsImage
PS>TerminatingError(Mount-WindowsImage): "The specified image in the specified wim is already mounted for read/write access.
"
Mount-WindowsImage : The specified image in the specified wim is already mounted for read/write access.

At C:\Program Files\WindowsPowerShell\Modules\OSD\25.2.23.1\Public\Functions\Dism\Mount-MyWindowsImage.ps1:140 char:13
+             Mount-WindowsImage @Params | Out-Null
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Mount-WindowsImage], COMException
    + FullyQualifiedErrorId : Microsoft.Dism.Commands.MountWindowsImageCommand
Mount-WindowsImage : The specified image in the specified wim is already mounted for read/write access.
```

Beyond that it starts adding the windows packages which all fail `WinPE-WMI.cab, WinPE.HTA.cab, etc.` 

Reasoning: `PS>TerminatingError(Add-WindowsPackage): "Cannot bind argument to parameter 'Path' because it is null."`

Then beyond that the saving of the windows image fails- due to the same error.

**Comment by OSDeploy** [comment]
I asked if you can make a first pass and narrow down the issue. If you read the Error Message then you have your answer.

i.e. it is failing because the WIM that it needs to mount is already mounted. Resolve this problem as it does not appear to be an issue with OSD.

**Comment by MichaelEscamilla** [comment]
Couple things you can try:

Run the below command to try and unmount any Images
```powershell
Get-WindowsImage -Mounted | Dismount-WindowsImage -Discard
```
Then try to create your template again.



If that does not work, check the below registry key for any mounted images. Delete them from here.
```
HKLM\SOFTWARE\Microsoft\WIMMount\mounted images\
```
Then run the below to clean up
```
dism /cleanup-wim
```
And then verify the images are unmounted with the below
```
dism /get-mountedwiminfo
```

**Comment by jkingster** [comment]
@MichaelEscamilla, manually deleting from registry and running the cleanup seems to have done the trick. Thanks for that. I've been stumped.
